src/nvim/ops.c

```diff
@@ -2512,10 +2504,11 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
   return;
 }
 
-static void yank_copy_line(yankreg_T *reg, struct block_def *bd, long y_idx)
+static void yank_copy_line(yankreg_T *reg, struct block_def *bd, size_t y_idx)
 {
-  char_u *pnew = xmallocz(bd->startspaces + bd->endspaces + bd->textlen);
-
+  assert(bd->startspaces + bd->endspaces + bd->textlen >= 0);
```

Comment:
indeed. A better way would be to assert the three separately but I think it's safe without. Removing this one.